src/main/webapp/app/exam/participate/exam-navigation-sidebar/exam-navigation-sidebar.component.ts

```diff
@@ -0,0 +1,254 @@
+import { SidebarCardDirective } from 'app/shared/sidebar/sidebar-card.directive';
+import { Subscription } from 'rxjs';
+import { SidebarEventService } from 'app/shared/sidebar/sidebar-event.service';
+import { SidebarData } from 'app/types/sidebar';
+import { ProfileService } from 'app/shared/layouts/profiles/profile.service';
+import { ExamSession } from 'app/entities/exam-session.model';
+import { Exercise, ExerciseType } from 'app/entities/exercise.model';
+import { ProgrammingSubmission } from 'app/entities/programming-submission.model';
+import { SubmissionVersion } from 'app/entities/submission-version.model';
+import { FileUploadSubmission } from 'app/entities/file-upload-submission.model';
+import { CodeEditorRepositoryService } from 'app/exercises/programming/shared/code-editor/service/code-editor-repository.service';
+import { CodeEditorConflictStateService } from 'app/exercises/programming/shared/code-editor/service/code-editor-conflict-state.service';
+import { ExamExerciseUpdateService } from 'app/exam/manage/exam-exercise-update.service';
+import { ButtonTooltipType, ExamParticipationService } from 'app/exam/participate/exam-participation.service';
+import { LayoutService } from 'app/shared/breakpoints/layout.service';
+import { map } from 'rxjs/operators';
+import { CustomBreakpointNames } from 'app/shared/breakpoints/breakpoints.service';
+import { CommitState, DomainChange, DomainType } from 'app/exercises/programming/shared/code-editor/model/code-editor.model';
+import { ProgrammingExercise } from 'app/entities/programming-exercise.model';
+import { IconProp } from '@fortawesome/fontawesome-svg-core';
+import { faChevronRight, faFileLines, faHourglassHalf } from '@fortawesome/free-solid-svg-icons';
+import { facSaveSuccess, facSaveWarning } from '../../../../content/icons/icons';
+import { getIconTooltip } from 'app/entities/exercise.model';
+
+export enum ExerciseButtonStatus {
+    Synced = 'synced',
+    SyncedSaved = 'synced saved',
+    NotSynced = 'notSynced',
+}
+
+@Component({
+    selector: 'jhi-exam-navigation-sidebar',
+    standalone: true,
+    imports: [ArtemisSidebarModule, ArtemisSharedModule, SidebarCardDirective],
+    templateUrl: './exam-navigation-sidebar.component.html',
+    styleUrl: './exam-navigation-sidebar.component.scss',
+})
+export class ExamNavigationSidebarComponent implements OnDestroy, OnInit {
+    @Input() sidebarData: SidebarData;
+    @Input() exercises: Exercise[] = [];
+    @Input() exerciseIndex = 0;
+    @Input() overviewPageOpen: boolean;
+    @Input() examSessions?: ExamSession[] = [];
+    @Input() examTimeLineView = false;
+    @Input() isTestRun = 0;
+    @Output() onPageChanged = new EventEmitter<{
+        overViewChange: boolean;
+        exercise?: Exercise;
+        forceSave: boolean;
+        submission?: ProgrammingSubmission | SubmissionVersion | FileUploadSubmission;
+    }>();
+
+    static itemsVisiblePerSideDefault = 4;
```

Comment:
Constants are usually in screaming snake case, I do also think that the default value can be readonly here? 
```suggestion
    static readonly ITEMS_VISIBLE_PER_SIDE_DEFAULT = 4;
```